Added a search box to the nav header for the mobile view.

shieldsjared opened the pull request:
Github Issue: #630 

** Recommend merging PR #664  before this one otherwise the UX will look FUNKY **

Added a magnifying glass icon to the top header, and centered the mattermost logo/header.  When clicking the magnifying glass, a searchbox appears.  Also, added safety catches in case the hamburger menu is already open (and vice versa).

![image](https://cloud.githubusercontent.com/assets/600643/20372187/67c516b4-ac37-11e6-9268-e5813476138d.png)

![image](https://cloud.githubusercontent.com/assets/600643/20372193/6e59ab70-ac37-11e6-879f-2aee9a57e208.png)

asaadmahmood commented:
If we can remove the dark border for the search bar on focus, that would be good.
![screen shot 2016-11-18 at 11 11 08 pm](https://cloud.githubusercontent.com/assets/11034289/20440994/513e9d8a-ade4-11e6-98da-b2c8a0a2b037.png)

+1 apart from that, looks good.

it33 commented:
@asaadmahmood actually, I'm wondering if the design is better with the dark border since it's more clear the box has focus--what's the alternative otherwise? 

Just to check, when you hit the search icon, is the cursor in the search box right away? If so, I think we're good to go. 

@asaadmahmood?

asaadmahmood commented:
@shieldsjared had a talk with @it33, we're going to remove the focus from the mobile experience, or if we want we can just have it slightly darker than the default border, the reason for that being that the element is already in focus on the screen on mobile, and there's a hint towards the focus when the keyboard pops up, which makes it quite apparent. Thus we don't really need that dark border on mobile for the focus state. Let me know if that sounds good.

shieldsjared commented:
I'm good with whatever :) I'll submit the update tomorrow.

shieldsjared commented:
Ok, @it33 and @asaadmahmood.  I removed the black border (wasnt there intentionally, just had to add some additional styles to the textbox to match the standard-view search box styling).  Also, added a call to ensure that when you press the magnifying glass, focus is automatically set on the textbox.

(commits squashed)

it33 commented:
I think Asaad's already done the dev review for this, given he's generated the output and noticed the black outline, merging as we seem to be blocking on the wrong PRs...

![image](https://cloud.githubusercontent.com/assets/177788/20868733/59ed9442-ba17-11e6-91a4-8d31eccc8cab.png)
